games/guess-it/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+#[derive(Debug, TypeId, Describe, Decode, Encode, Clone, Serialize, Deserialize)]
+pub struct Player {
+    pub guess: u128,
+}
+impl Player {
+    pub fn empty() -> Player {
+        return Self {
+            guess: 0u128,
+        }
+    }
+
+    pub fn get_guess(&self) -> u128 { self.guess }
+}
+
+#[derive(TypeId, Describe, Decode, Encode)]
+#[derive(NonFungibleData)]
+struct PlayerNft {
+    name: String,
+}
+
+blueprint! {
+    struct GuessIt {
+        bank: Vault,
+        badges: Vault,
+        badge_ref: ResourceDef,
+        game: Game,
+    }
+
+    impl GuessIt {
+        pub fn create(name: String, bet: u128) -> Component {
+            let player_badges: Bucket = ResourceBuilder::new_non_fungible()
+                .metadata("name", "Guess-It Game")
+                .metadata("symbol", "GIG")
+                .initial_supply_non_fungible([
+                    (
+                        NonFungibleKey::from(Uuid::generate()),
+                        PlayerNft {name: "Player 1".to_string()},
+                    ),
+                    (
+                        NonFungibleKey::from(Uuid::generate()),
+                        PlayerNft {name: "Player 2".to_string()},
+                    ),
+                ]);
+            let player_badge_def = player_badges.resource_def();
+
+            Self {
+                bank: Vault::new(RADIX_TOKEN),
+                badges: Vault::with_bucket(player_badges),
+                badge_ref: player_badge_def,
+                game: Game::new(None, name, bet),
+            }.instantiate()
+        }
+
+        pub fn join(&mut self, mut payment: Bucket) -> (Bucket, Bucket, String) {
+            assert_eq!(self.game.state, State::AcceptingChallenger, "Not accepting challengers!");
+            assert!(self.game.players.len() <= 2, "Too many players!");
+
+            let player: Player = Player::empty();
+            let badge: Bucket = self.badges.take(Decimal::one());
+            let nft_id: NonFungibleKey = badge.get_non_fungible_key();
+            // Secure the funds
+            self.bank.put(payment.take(Decimal::from(self.game.bet as i128)));
+            // Add the player's details
+            self.game.players.insert(nft_id.clone(), player.clone());
+            self.game.state = if self.game.players.len() == 2 { State::MakeGuess } else { State::AcceptingChallenger };
+
+            return (badge, payment, nft_id.to_string());
+        }
+
+        pub fn state(&mut self) -> String {
+            Game::serialize(&self.game)
+        }
+
+        #[auth(badge_ref)]
+        pub fn make_guess(&mut self, guess: u128) -> String {
+            assert_eq!(self.game.state, State::MakeGuess, "Not ready to make a guess!");
+
+            let key = auth.get_non_fungible_key();
+            let player = self.game.players.get_mut(&key).unwrap();
+            player.guess = guess;
+
+            self.game.state = if self.game.has_guessed() { State::WinnerSelection } else { State::MakeGuess };
```

Comment:
Just style preference here, i really like using match:
```rust
self.game.state = match self.game.has_guessed() {
  true => State::WinnerSelection,
  false => State::MakeGuess 
};
```

Or maybe you can move this to the `Game`, namely `self.game.get_state()`

Very minor though

Reply:
honestly was thinking of a way to use match.

Will update next commit

Reply:
I will review the changes you make in the next series of commits and then merge it.